ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,109 @@ actor_system_config:
 | `resolution` | When making a schedule for sending messages, discrete time slots are used. The slot duration is set by the `resolution` parameter in microseconds. |
 
+## Memory controller {#memory-controller}
+
+There are many components inside {{ ydb-short-name }} [database nodes](../../concepts/glossary.md#database-node) that utilize memory. Most of them need a fixed amount, but some are flexible and can use varying amounts of memory, typically to improve performance. If such components allocate more memory than is physically available, the operating system is likely to [terminate](https://en.wikipedia.org/wiki/Out_of_memory#Recovery) the entire {{ ydb-short-name }} node, which is undesirable. The memory controller's goal is to allow {{ ydb-short-name }} to avoid out-of-memory situations while still efficiently using the available memory.
+
+Examples of components managed by the memory controller:
+
+- [Shared cache](../../concepts/glossary.md#shared-cache): stores recently accessed data pages read from [distributed storage](../../concepts/glossary.md#distributed-storage) to reduce disk I/O and accelerate data retrieval.
+- [MemTable](../../concepts/glossary.md#memtable): holds data that has not yet been flushed to [SST](../../concepts/glossary.md#sst).
+- [KQP](../../concepts/glossary.md#kqp): stores intermediate query results.
+- Allocator caches: keep memory blocks that have been released but not yet returned to the operating system.
+
+Memory limits can be configured to control overall memory usage, ensuring the database operates efficiently within the available resources.
+
+### Hard memory limit {#hard-memory-limit}
+
+The hard memory limit specifies the total amount of available memory.
+
+By default, the hard memory limit is set to the {{ ydb-short-name }} node process's [cgroups](https://en.wikipedia.org/wiki/Cgroups) memory limit.
+
+In environments without a cgroups memory limit, the default hard memory limit is equal to the host's total available memory. This allows the database to utilize all available resources in unrestricted environments, though it may lead to resource contention with other processes running on the same host.
+
+Additionally, the hard memory limit can be specified in the configuration. Note that the database process may still exceed this limit, so it is highly recommended to use cgroups memory limits in production environments to enforce strict memory control.
+
+Example of the `memory_controller_config` section with a specified hard memory limit:
+
+```yaml
+memory_controller_config:
+  hard_limit_bytes: 16106127360
+```
+
+### Soft memory limit {#soft-memory-limit}
```

Comment:
You do treat "soft memory limit" for shared cache only, which is confusing... Do you have "soft memory limit" term for KQP? Why?

Reply:
Soft memory limit (and target utilization) works in a context of a whole YDB process

It's just that size of Shared Cache (and MemTable) calculated by evaluating current YDB process memory consumption to these values

Reply:
Added

> The soft memory limit specifies a dangerous threshold that should not be exceeded by **a YDB process** under normal circumstances.